parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -868,6 +868,24 @@ providers:
     apiVersion: credentialprovider.kubelet.k8s.io/v1
     args:
       - /etc/kubernetes/azure.json
+EOF
+    elif [[ -n "${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER}" ]]; then
+        tee "${CREDENTIAL_PROVIDER_CONFIG_FILE}" > /dev/null <<EOF
+apiVersion: kubelet.config.k8s.io/v1
+kind: CredentialProviderConfig
+providers:
+  - name: acr-credential-provider
+    matchImages:
+      - "*.azurecr.io"
+      - "*.azurecr.cn"
+      - "*.azurecr.de"
+      - "*.azurecr.us"
+      - "mcr.microsoft.com"
+    defaultCacheDuration: "10m"
+    apiVersion: credentialprovider.kubelet.k8s.io/v1
+    args:
+      - /etc/kubernetes/azure.json
+      - --registry-mirror=mcr.microsoft.com:$BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER
 EOF
     else
     tee "${CREDENTIAL_PROVIDER_CONFIG_FILE}" > /dev/null <<EOF
```

Comment:
nit, not related to this PR, but that will be great if you can add an intent
```suggestion
        tee "${CREDENTIAL_PROVIDER_CONFIG_FILE}" > /dev/null <<EOF
```

Reply:
fixed.